File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(vector<vector<string>> board, int h, int w) {
    int answer = 0;
    
    int dh[4] = {0, 1, -1, 0};
    int dw[4] = {1, 0, 0, -1};
    
    for(int i = 0; i < 4; i++){
        if ((0 <= h + dh[i]) && (h + dh[i] < board.size()) && (0 <= w+dw[i]) && (w+dw[i] < board[0].size())){
            if (board[h][w] == board[h+dh[i]][w+dw[i]]){
                answer++;
            }
        }
    }
    
    return answer;
}

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.py
def solution(board, h, w):
    answer = 0
    count = 0
    n = len(board)
    dh = [0, 1, -1, 0]
    dw = [1, 0, 0, -1]
    for i in range(4):
        h_check = h + dh[i]
        w_check = w + dw[i]
        if(n > h_check >= 0 and n > w_check >= 0):
            if(board[h][w] == board[h_check][w_check]):
                count+=1
    return count